### PR TITLE
Nist MS Search Update

### PR DESCRIPTION
**Pull request by Kevin-Murray**
Update to NIST MS Search identification module.

Made module compatible with MS/MS, clustered spectra produced from spectral deconvolution modules, and precursor ion searches with the NIST MS Search program.  Added optional MS/MS merging and integer mz conversion functionality. 

Removed ionization method, spectrum RT tolerance, max peaks and identity check parameters and all functions dependent on these parameters. 

Updated module would not be compatible with legacy workflows dependent on this module.

### CODE REVIEW

**Comment by tomas-pluskal** [comment]
Thanks Kevin, the code looks good. I'll take a closer look at the diff, but currently there is a lot of formatting changes. Could you please use the code formatter that we provide (mzmine-eclipse-code-formater.xml or mzmine-intellij-code-formater.xml) and reformat your code?

**Comment by Kevin-Murray** [comment]
I think I successfully applied the code formatter. I have never used one before. Let me know if it still incorrectly formatted and I will try a second attempt.

File: src/main/java/io/github/mzmine/modules/dataprocessing/id_nist/NistMsSearchTask.java [hunk 1]
```diff
@@ -706,4 +603,44 @@ private File getSecondLocatorFile(final File primaryLocatorFile) throws IOExcept
...
+   * Converts DataPoint ions mz to int.
+   *
+   * Function adapted from module: adap.mspexport.
+   *
+   * @param dataPoints spectra to convert.
+   * @param mode       conversion method.
+   * @return DataPoint array converted to integers.
+   * @throws IOException if there are i/o problems.
+   */
+  private DataPoint[] integerDataPoints(final DataPoint[] dataPoints, final String mode) {
```
Comment (tomas-pluskal) [inline src/main/java/io/github/mzmine/modules/dataprocessing/id_nist/NistMsSearchTask.java:617]: This function would be better moved into some utility class, like ScanUtils (in io.github.mzmine.util.scans). Then both adap.msexport and this module can share the same utility method.

File: src/main/java/io/github/mzmine/modules/dataprocessing/id_nist/NistMsSearchTask.java [hunk 2]
```diff
@@ -706,4 +603,44 @@ private File getSecondLocatorFile(final File primaryLocatorFile) throws IOExcept
...
+
+    for (int i = 0; i < size; ++i) {
+      double mz = (double) Math.round(dataPoints[i].getMZ());
+      double intensity = dataPoints[i].getIntensity();
+      Double prevIntensity = integerDataPoints.get(mz);
+      if (prevIntensity == null) {
+        prevIntensity = 0.0;
+      }
+
+      if ("Merging mode: Sum".equals(mode)) {
```
Comment (tomas-pluskal) [inline src/main/java/io/github/mzmine/modules/dataprocessing/id_nist/NistMsSearchTask.java:631]: Please don't use the same String constants like this in multiple places of the code. Create an enum for them, or a final String[] array.

File: gradle.properties [hunk 3]
```diff
@@ -0,0 +1,12 @@
+netbeans.de-funfried-netbeans-plugins-externalcodeformatter.enableIndentationSettings=true
```
Comment (tomas-pluskal) [inline gradle.properties:1]: Please remove the changes to gradle.properties from the pull request

File: src/main/java/io/github/mzmine/modules/dataprocessing/id_nist/NistMsSearchTask.java (hunk 1 above)
Comment (tomas-pluskal) [inline src/main/java/io/github/mzmine/modules/dataprocessing/id_nist/NistMsSearchTask.java:617]: Create an enum for the `mode` parameter

**Comment by tomas-pluskal** [comment]
Thanks Kevin, the formatting is now fixed, I added a few additional comments.

**Comment by Kevin-Murray** [comment]
Thanks for the feedback Tomas.

I made all the changes. I added the new ScanUtils integer conversion method to the ADAP MSP/MGF export modules.

**Comment by tomas-pluskal** [comment]
Hi Kevin, thanks!
I noticed you removed the gradle.properties file. You should not remove it, it is essential for MZmine. I just wanted you to remove the *modifications* to the file from the pull request.

**Comment by tomas-pluskal** [comment]
Nevermind, I see that there are modifications to gradle.properties in the current pull request. I think this is ready for merging now.

**Pull request merged**